web/src/app/Connection.js

```diff
@@ -1,7 +1,14 @@
+/* eslint-disable max-classes-per-file */
 import { basicAuth, bearerAuth, encodeBase64Url, topicShortUrl, topicUrlWs } from "./utils";
 
 const retryBackoffSeconds = [5, 10, 20, 30, 60, 120];
 
+export class ConnectionState {
+  static Connected = "connected";
+
+  static Connecting = "connecting";
+}
+
```

Comment:
Can you explain why you move this up?

Reply:
This rule is generally just to avoid any possible gotchas with JS' hoisting behaviour. So the moving was to satisfy having everything defined before its use.

- <https://github.com/airbnb/javascript#hoisting>
- <https://eslint.org/docs/latest/rules/no-use-before-define>
- <https://developer.mozilla.org/en-US/docs/Glossary/Hoisting>